python/pylibcugraph/pylibcugraph/triangle_count.pyx

```diff
@@ -0,0 +1,137 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Have cython use python 3 syntax
+# cython: language_level = 3
+
+from libc.stdint cimport uintptr_t
+
+from pylibcugraph._cugraph_c.resource_handle cimport (
+    bool_t,
+    data_type_id_t,
+    cugraph_resource_handle_t,
+)
+from pylibcugraph._cugraph_c.error cimport (
+    cugraph_error_code_t,
+    cugraph_error_t,
+)
+from pylibcugraph._cugraph_c.array cimport (
+    cugraph_type_erased_device_array_view_t,
+    cugraph_type_erased_device_array_view_create,
+    cugraph_type_erased_device_array_view_free,
+)
+from pylibcugraph._cugraph_c.graph cimport (
+    cugraph_graph_t,
+)
+from pylibcugraph._cugraph_c.community_algorithms cimport (
+    cugraph_triangle_count_result_t,
+    cugraph_triangle_count,
+    cugraph_triangle_count_result_get_vertices,
+    cugraph_triangle_count_result_get_counts,
+    cugraph_triangle_count_result_free,
+)
+from pylibcugraph.resource_handle cimport (
+    ResourceHandle,
+)
+from pylibcugraph.graphs cimport (
+    _GPUGraph,
+)
+from pylibcugraph.utils cimport (
+    assert_success,
+    copy_to_cupy_array,
+    assert_CAI_type,
+    get_c_type_from_numpy_type,
+)
+
+
+def EXPERIMENTAL__triangle_count(ResourceHandle resource_handle,
+                                 _GPUGraph graph,
+                                 start_list,
+                                 bool_t do_expensive_check):
+    """
+    Computes the number of triangles (cycles of length three) and the number
+    per vertex in the input graph.
+
+    Parameters
+    ----------
+    resource_handle: ResourceHandle
+        Handle to the underlying device and host resources needed for
+        referencing data and running algorithms.
+
+    graph: MGGraph
+        The input graph, for Multi-GPU operations.
+
+    start_list: device array type
```

Comment:
It looks like from the code that `start_list` can also be `None`.  Can you describe the behavior here if the user passes `None` for `start_list`?

Reply:
My understanding is if `start_list=None`, `triangle_counts` will compute the number of triangle for each vertex in the graph as opposed to a subset of the graph if `start_list !=None` like NetworkX